### PR TITLE
Adding ASCII_Subsequence in C, Python and JavaScript (Closes issue #1703)

### LINKED ISSUE #1703: Adding ASCII Subsequence in C,Python and JavaScript
I would love to contribute to this project. I am a GSSOC'20 participant and I am very excited to work upon this issue. Please assign me this issue.

### PR DESCRIPTION
**Pull request by Aman-Codes**
Fixes #1703

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- Added ASCII_Subsequences in C,Python and JavaScript
- Removed Duplicate files

#### Languages Used:
- C
- Python
- JavaScript

#### Files Added:
- ASCII_Subsequences.c
- ASCII_Subsequences.py
- ASCII_Subsequences.js

@jainaman224 @raghav-dalmia @asha15 @singh-shreya6

### CODE REVIEW

**Comment by Aman-Codes** [comment]
Please review it.
@jainaman224 @raghav-dalmia @asha15 @singh-shreya6

File: Ascii_Subsequences/ASCII_Subsequences.c [hunk 1]
```diff
@@ -0,0 +1,114 @@
+/*ASCII SUBSEQUENCES PROBLEM
+This problem is same as the "Subsequences" question just instead of two function calls, three function calls will be made :
+- One with the same string.
+- One with a letter added.
+- One with the ASCII code of the letter added to the string.
+- For example:-
+Take as input str, a string. We are concerned with all the possible ASCII-subsequences of str. E.g. “ab” has following ASCII-subsequences “”, “b”, “98”, “a”, “ab”, “a98”, “97”, “97b”, “9798”
+a. Write a recursive function which returns the count of ASCII-subsequences for a given string. Print the value returned.
+b. Write a recursive function which prints all possible ASCII-subsequences for a given string (void is the return type for function).
```
Comment (asha15) [inline Ascii_Subsequences/ASCII_Subsequences.c]: one line should contain max 80 characters. please change the comment according to it
a brief introduction is enough

File: Ascii_Subsequences/ASCII_Subsequences.c [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+- One with the ASCII code of the letter added to the string.
+- For example:-
+Take as input str, a string. We are concerned with all the possible ASCII-subsequences of str. E.g. “ab” has following ASCII-subsequences “”, “b”, “98”, “a”, “ab”, “a98”, “97”, “97b”, “9798”
+a. Write a recursive function which returns the count of ASCII-subsequences for a given string. Print the value returned.
+b. Write a recursive function which prints all possible ASCII-subsequences for a given string (void is the return type for function).
+*/
+//*********************//
+/*Time Complexity:
+O(3^N) where N = String Length.*/
+//*******************//
```
Comment (asha15) [inline Ascii_Subsequences/ASCII_Subsequences.c]: remove this

File: Ascii_Subsequences/ASCII_Subsequences.c [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <math.h>
+
+#define SIZE 100
+
```
Comment (asha15) [inline Ascii_Subsequences/ASCII_Subsequences.c]: remove this line

File: Ascii_Subsequences/ASCII_Subsequences.c [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    int i1;
+    char s2[strlen(s) + 2];
+
+    for(i1 = 0; i1 < strlen(s); i1++)
+    {
+        s2[i1] = s[i1]; // Copy string from s to s2
+    }
+
+    s2[i1++] = str[i]; // Concatinate ith character of string str to s2
```
Comment (asha15) [inline Ascii_Subsequences/ASCII_Subsequences.c]: take the comment into the previous line

File: Ascii_Subsequences/ASCII_Subsequences.c [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    s2[i1] = '\0';
+
+    subseq(s2, i + 1, str); // Second function call
+
+    int i2, n = (int)str[i]; // Integer n stores the ASCII value of the ith character of string str
```
Comment (asha15) [inline Ascii_Subsequences/ASCII_Subsequences.c]: take the comment into the previous line

**Comment by Aman-Codes** [comment]
Did the changes asked.
@jainaman224 @asha15 @raghav-dalmia @singh-shreya6

File: Ascii_Subsequences/ASCII_Subsequences.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+	}
+    
+	// The 3 function calls in the same order as explained above.
+	// First function call
+	subseq(s, i + 1, str); 
+	// Second function call
+	subseq(s + str[i], i + 1, str); 
+	var n = str.charCodeAt(i); 
+	// Third function call with the ASCII value of the ith element of the string
+	subseq(s + n.toString(), i + 1, str);
```
Comment (asha15) [inline Ascii_Subsequences/ASCII_Subsequences.js]: correct the indentation

File: Ascii_Subsequences/ASCII_Subsequences.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    
+	// The 3 function calls in the same order as explained above.
+	// First function call
+	subseq(s, i + 1, str); 
+	// Second function call
+	subseq(s + str[i], i + 1, str); 
+	var n = str.charCodeAt(i); 
+	// Third function call with the ASCII value of the ith element of the string
+	subseq(s + n.toString(), i + 1, str); 
+
```
Comment (asha15) [inline Ascii_Subsequences/ASCII_Subsequences.js]: remove this line

File: Ascii_Subsequences/ASCII_Subsequences.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+( function IIFE() {
+    var str = prompt("Enter the string");
+	console.log( "Number of Subsequences : ", Math.pow(3, str.length));
+	subseq("", 0, str);
+	return 0;
```
Comment (asha15) [inline Ascii_Subsequences/ASCII_Subsequences.js]: correct the indentation

**Comment by asha15** [comment]
squash your commits and push it

**Comment by Aman-Codes** [comment]
Changes done ma'am pleas review it.
@jainaman224 @raghav-dalmia @asha15 @singh-shreya6

**Comment by Aman-Codes** [comment]
Thank you ma'am.

**Pull request merged**